sympy/solvers/bivariate.py

```diff
@@ -114,6 +120,45 @@ def _linab(arg, symbol):
     return a, b, x
 
 
+def power_list(f, *symbols):
+    """
+    Helper function to return powers of variables
+    present in f in dictionary.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The function whose powers list is to be returned.
+
+    symbols : Symbol
+        The symbols whose powers are to be returned.
+
+    Returns
+    =======
+
+    Powers of symbols in f in dictionary form.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y
+    >>> from sympy.solvers.bivariate import power_list
+    >>> power_list(x**2 + x*y + 1, x)
+    [(2,), (1,)]
+    >>> power_list(x**2 + x*y + 1, x, y)
+    [(2,), (1,), (1,)]
+    """
+    result = []
+    for t in tf(f):
+        if isinstance(t, (trigs, Abs, log, exp)):
+            result += power_list(t.args[0], *symbols)
+        elif t.has(*symbols):
+            p = t.as_poly()
+            result += p.monoms()
+    return result
```

Comment:
> Also can you try getting result seperately for each symbol specified in the parameter, say in a dictionary. This will look more convincing.

I tried to use the returning pattern of `p.monoms()` but can't achieve anything. And this improvement can also be done in a separate PR i think.

Reply:
> And this improvement can also be done in a separate PR i think.

I think this is very much the part of this PR, It's not in the minimal usable state.